py/desisim/io.py

```diff
@@ -387,24 +388,100 @@ def read_simspec_mpi(filename, comm, spectrographs=None):
     # Note: this is global scope within the function, so memmap file handle
     # will not close until the function returns (which is fine).
     hdus = None
+    hdus = fits.open(filename, memmap=True)
```

Comment:
Please update so that only rank 0 will open the fits file, broadcast the shapes to all the other ranks, and then proceed with broadcasting the data.  As currently written, all ranks will open the same file and scan its headers to get the shapes, potentially bogging down the I/O server.

Otherwise these changes look fine.  Thanks.

Reply:
Good catch. Changes made, some re-arranging was required.